### PR TITLE
Replace CAF 18.6 deprecated methods in favour of supported free function

### PR DESCRIPTION
**Pull request by patszt**
CAF 18.6 removed the make_sink, make_source and make_continuous_source methods.
The free functions i have replaced them with are supported both in CAF 17.6 and 18.6
### :memo: Reviewer Checklist

Review this pull request by ensuring the following items:

- [x] All user-facing changes have changelog entries
- [x] User-facing changes are reflected on [vast.io](https://github.com/tenzir/vast/tree/master/web)

### CODE REVIEW

File: libvast/native-plugins/segment_store.cpp [hunk 1]
```diff
@@ -190,7 +191,7 @@ system::store_actor::behavior_type passive_local_store(
         if (!seg) {
           VAST_ERROR("{} couldn't create segment from chunk: {}", *self,
                      seg.error());
-          self->send_exit(self, caf::exit_reason::unhandled_exception);
```
Comment (patszt) [inline libvast/native-plugins/segment_store.cpp:193]: in CAF 18.6 this enum value is removed

Comment (dominiklohmann) [reply libvast/native-plugins/segment_store.cpp:193]: Note that this doesn't have to be a `caf::exit_reason`, you can actually provide any error code here. I think it'd make a lot of sense to just use `caf::make_error(vast::ec::something, fmt::format("{} failed to create segment from chunk: {}", *self, seg.error()))` here.

Comment (patszt) [reply libvast/native-plugins/segment_store.cpp:193]: @dominiklohmann 
Good tip, i will change it as proposed

**Review by dominiklohmann — APPROVED** [review]
This is a no-brainer refactoring, thanks!

**Pull request merged**